lib/packwerk/extracting_references/README.md

```diff
@@ -0,0 +1,4 @@
+# Extracting references
+
+- Given a `AST::Node`, this module extracts `Packwerk::Reference`.
```

Comment:
If you put it like this, it sounds like everything here works in the context of a single node. However, `ParsedConstantDefinitions` [works on a whole syntax tree](https://github.com/Shopify/packwerk/pull/144/files#diff-8538012fa017b7f4e422be58e6e8adbda8cf1fdba6bbdd45df7b41d8ad5afa64R7).